MM-15161 show bot badge and icon for ephemeral posts from bots

scottleedavis opened the pull request:
#### Summary

This adds the profile image, username and bot badge to ephemeral posts for bots.

#### Ticket Link

Fixes https://github.com/mattermost/mattermost-server/issues/10640


#### Related Pull Requests
- Has mobile changes https://github.com/mattermost/mattermost-mobile/pull/2740

mattermod commented:
`Setup Test Server` label detected. Spinmint test server created if build succeeds (checks pass and no conflicts with base branch).

hanzei commented:
@scottleedavis Do you have a version of the remind plugin that can be used to test this?

mattermod commented:
Spinmint test server created at: https://i-0e4939484f312b71d.test.spinmint.com

Test Admin Account: Username: `sysadmin` | Password: `sysadmin`

Test User Account: Username: `user-1` | Password: `user-1`

Internal IP: 172.26.46.128
Instance ID: i-0e4939484f312b71d

scottleedavis commented:
> @scottleedavis Do you have a version of the remind plugin that can be used to test this?

@hanzei  sort of.  If you use the ' ephemeral-bot-posts' branch on the repo, and thing typed into `/remind ....` will return an ephemeral and a normal test posts.

<img width="402" alt="Screen Shot 2019-04-25 at 2 22 36 AM" src="https://user-images.githubusercontent.com/2914443/56727504-a8713680-6705-11e9-8855-0d512e14f60d.png">

hanzei commented:
Great, thanks Scott!

jasonblais commented:
Thank you! It appears to be working for me

![image](https://user-images.githubusercontent.com/13119842/56735019-5e527a00-6732-11e9-9c71-78c453012e9f.png)

Is there any additional testing I should take into account?

mattermod commented:
Spinmint test server destroyed

saturninoabril commented on components/post_profile_picture/post_profile_picture.jsx:
```diff
@@ -70,7 +70,7 @@ export default class PostProfilePicture extends React.PureComponent {
         const isSystemMessage = PostUtils.isSystemMessage(post);
         const fromWebhook = PostUtils.isFromWebhook(post);
-        if (isSystemMessage && !compactDisplay && !fromWebhook) {
+        if (isSystemMessage && !compactDisplay && !fromWebhook && !user.is_bot) {
```
Looks like `user` needs to check if not null since it's not required in this component.  Might also good to do the same as [post_header index](https://github.com/mattermost/mattermost-webapp/blob/e8485d29d6dfefd4900d2a7e71ab11d6af1a74a7/components/post_view/post_header/index.js#L14) where it passed `isBot` prop.

scottleedavis replied:
Changed both to use the ternary operator.  Please let me know if this is what was requested.

scottleedavis replied:
Do the latest changes match what you requested @saturninoabril ?

saturninoabril replied:
Thanks, matched with the latest push.

scottleedavis commented:
@saturninoabril I rebased off of master, and then I noticed a lot appears now changed in this PR beyond the two small changes.

 I am use to keeping a PR/MR up to date with master and that is what I thought I had done, and this has not happened before to me. 

Please advise the best course of action.  I apologize.

saturninoabril commented on components/post_profile_picture/index.jsx:
```diff
@@ -10,12 +10,14 @@ import PostProfilePicture from './post_profile_picture';
         status: getStatusForUserId(state, ownProps.userId),
-        user: getUser(state, ownProps.userId),
+        isBot: Boolean(user && user.is_bot),
+        user: user
```
nit,
```suggestion
        user
```

levb commented:
@scottleedavis looks like you need to merge from master, and re-run the tests. Do you mind doing it so we can merge?

hanzei commented:
@svelle I think this can be merged anyway. It has no merge conflicts.

scottleedavis commented:
Done @levb

hanzei commented:
Thank you very much for this PR @scottleedavis!
